notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/NotificationEventConfig.java

```diff
@@ -0,0 +1,205 @@
+    public KafkaSetupHandler kafkaSetupHandler(NewTopicProperties newTopicProperties) {
+        Objects.requireNonNull(notificationEventsTopic, getWarningMessageOnNotDefinedProp("enabled notification events", KAFKA_TOPIC_NAME_PROPERTY));
+        Objects.requireNonNull(bootstrapServers, getWarningMessageOnNotDefinedProp("enabled notification events", KAFKA_BOOTSTRAP_SERVERS_PROPERTY));
+
+        return KafkaSetupHandler.builder()
+                .newTopic(Optional.of(isKafkaSetupEnabled).filter(Boolean.TRUE::equals).map(enabled -> newTopicProperties.buildNewTopic(notificationEventsTopic)))
+                .notificationEventsTopic(notificationEventsTopic)
+                .bootstrapServers(bootstrapServers)
+                .isKafkaSetupEnabled(isKafkaSetupEnabled)
+                .build();
+    }
+
+    @Data
+    @ConfigurationProperties(prefix = KAFKA_TOPIC_SETUP_PROPERTY)
+    public static class NewTopicProperties {
+        private Integer numPartitions;
+        private Short replicationFactor;
+        private Map<String, String> configs = null;
+
+        public NewTopic buildNewTopic(final String topicName) {
+            // We execute this method and its validations only when 'notification.events.kafka.topic.setup' is true
+            // during KafkaSetupHandler building. If we use @Validated and its constraints, bean loading will fail even
+            // when topic setup is disabled.
+            final String component = "enabled Kafka topic setup";
+
+            Objects.requireNonNull(topicName, getWarningMessageOnNotDefinedProp(component, KAFKA_TOPIC_NAME_PROPERTY));
+            Objects.requireNonNull(numPartitions, getWarningMessageOnNotDefinedProp(component,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".numPartitions")));
+            Objects.requireNonNull(replicationFactor, getWarningMessageOnNotDefinedProp(component,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".replicationFactor")));
+
+            final String gtZeroWarning = " must be greater than zero";
+
+            Preconditions.checkArgument(numPartitions.compareTo(0) > 0,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".numPartitions").concat(gtZeroWarning));
+            Preconditions.checkArgument(replicationFactor.intValue() > 0,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".replicationFactor").concat(gtZeroWarning));
+
+            return new NewTopic(topicName, numPartitions, replicationFactor).configs(configs);
+        }
+    }
+
+    @Data
+    @ConfigurationProperties(prefix = CUSTOM_SCHEMA_TYPE_PREFIX)
+    public static class SchemaParserProperties {
+        private Boolean customEnabled;
+        private String keyParserClass;
+        private String keyParserMethod;
+        private String valueParserClass;
+        private String valueParserMethod;
+
+        public Function<Schema, ?> buildSchemaToKeyRecord() {
+            return Optional.ofNullable(customEnabled)
```

Comment:
I'm not really a fan of the `Optional<Boolean>` pattern but I can see the appeal.

Prefer `Boolean::booleanValue` over `Boolean.TRUE::equals`

Reply:
Sure thing (added to the modification list)

Reply:
Fixed